### PR TITLE
Develop upstream sync 231219

### PR DESCRIPTION
**Pull request by zoranjovanovic-ns**

### CODE REVIEW

**Comment by draganmladjenovic** [comment]
Run Ubuntu-CPU please.

**Comment by draganmladjenovic** [comment]
Retest Ubuntu-CPU please.

**Comment by zoranjovanovic-ns** [comment]
Retest Ubuntu-CPU please.

**Comment by zoranjovanovic-ns** [comment]
Hi @jayfurmanek, it seems that all CI builders except local_xla finished successfully.

**Comment by jayfurmanek** [comment]
gpu-pycpp still has fails. I fixed that but the fix ended up only in the rocm6 build container.  The PR check is still 5.7. I'll kick off a one-off for that for 5.7 as well then we can re-kick that one to be sure. Looks good otherwise.

**Comment by zoranjovanovic-ns** [comment]
Let me know if I need to do something.

**Comment by jayfurmanek** [comment]
retest gpu-pycpp please

File: third_party/xla/xla/service/gpu/ir_emitter.cc [hunk 1]
```diff
@@ -263,14 +263,7 @@ void IrEmitter::BindFusionArguments(const HloInstruction* fusion,
 void IrEmitter::MaybeEmitFenceForAMDGPU(llvm::AtomicOrdering atomic_ordering,
                                         const char* sync_scope_id) {
   if (IsEmittingForAMDGPU() &&
-<<<<<<< HEAD
-      (ir_emitter_context_->rocm_compute_capability().gcn_arch_name().substr(0, 6) == "gfx90a" || 
-       ir_emitter_context_->rocm_compute_capability().gcn_arch_name().substr(0, 6) == "gfx940" || 
-       ir_emitter_context_->rocm_compute_capability().gcn_arch_name().substr(0, 6) == "gfx941" || 
-       ir_emitter_context_->rocm_compute_capability().gcn_arch_name().substr(0, 6) == "gfx942")) {
-=======
       ir_emitter_context_->rocm_compute_capability().fence_before_barrier()) {
```
Comment (jayfurmanek) [inline third_party/xla/xla/service/gpu/ir_emitter.cc:266]: @wenchenvincent , @i-chaochen, @pemeliya Is this the proper way to check if the barrier is needed now?

Comment (i-chaochen) [reply third_party/xla/xla/service/gpu/ir_emitter.cc:266]: Yes, I think `@local_xla//xla/stream_executor/gpu:gpu_kernel_test_gpu` is for this one, espeically on `GpuKernelTilingTest.ColReductionWithSmallDtype` 


see the full details : https://github.com/ROCmSoftwarePlatform/frameworks-internal/issues/3654#issuecomment-1446838496 and https://github.com/ROCmSoftwarePlatform/frameworks-internal/issues/5866

**Comment by i-chaochen** [comment]
http://ml-ci.amd.com:21096/blue/rest/organizations/jenkins/pipelines/tensorflow/pipelines/ThirdParty-XLA/branches/PR-2336/runs/3/nodes/41/steps/45/log/?start=0

Regarding the results of local xla, @zoranjovanovic-ns  could you try this PR for topk? https://github.com/openxla/xla/pull/7905

Also, do you know the full details of failed `gpu_kernel_test_gpu`? I am not sure whether it's related to fence check in the above. 

And I think `xla/tests:local_client_execute_test_gpu` should not be failed. (it's PASSED on the [12-19](http://ml-ci.amd.com:21096/job/tensorflow/job/OpenXLA/job/XLA-Internal-multibranch/job/ci_build_brake_fix_231219/1/consoleFull) and [12-21](http://ml-ci.amd.com:21096/job/tensorflow/job/OpenXLA/job/XLA-Internal-multibranch/job/ci_build_brake_fix_231221/1/consoleFull) fixed CI)

File: third_party/xla/xla/service/gpu/ir_emitter.cc (hunk 1 above)
Comment (jayfurmanek) [reply third_party/xla/xla/service/gpu/ir_emitter.cc:266]: yeah these are what I was looking for:
https://github.com/openxla/xla/pull/7277
https://github.com/openxla/xla/commit/2fce0d6c94ad1fdba5def0100ca5bd85b7397bd6

should be good here

**Comment by draganmladjenovic** [comment]
@i-chaochen I'll check gpu_kernel_test_gpu. According to pemeliya it fails on XLA too.

**Comment by i-chaochen** [comment]
> @i-chaochen I'll check gpu_kernel_test_gpu. According to pemeliya it fails on XLA too.

Yes, because there are some code in our develop-upstream in gpu_kernel_test_gpu hasn't upstreamed. So it's failed in XLA upstream but it's PASSED on develop-upstream before. https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/2298

**Review by jayfurmanek — APPROVED** [review]

**Comment by zoranjovanovic-ns** [comment]
Hi @i-chaochen I will try given PR for topk.
Also, will take a look at xla/tests:local_client_execute_test_gpu

**Comment by zoranjovanovic-ns** [comment]
I have created PR (just for testing purposes) with commits from https://github.com/openxla/xla/pull/7905 :

https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/2342

Test topk_kernel_test PASS now.

**Comment by i-chaochen** [comment]
@zoranjovanovic-ns thanks a lot for the checking! 

Since this weekly-sync is merged, we could use your test PR (based on topK fixed) to merge our develop-upstream? 

Also I notice that ```xla/tests:local_client_execute_test_gpu``` is also PASSED on that PR, these two could be related to I guess.

**Pull request merged**